usecases/schema/update.go

```diff
@@ -269,22 +270,55 @@ func (m *Manager) CreateSnapshot(ctx context.Context, principal *models.Principa
 func (m *Manager) RestoreSnapshot(ctx context.Context, principal *models.Principal,
 	className, storageName, ID string,
 ) (*models.SnapshotRestoreMeta, error) {
+	m.RestoreStatus = models.SnapshotRestoreMetaStatusSTARTED
 	path := fmt.Sprintf("schema/%s/snapshots/%s/%s/restore", className, storageName, ID)
 	if err := m.authorizer.Authorize(principal, "restore", path); err != nil {
 		return nil, err
 	}
+	go func(ctx context.Context, className, snapshotId string) {
+		m.RestoreStatus = models.SnapshotRestoreMetaStatusTRANSFERRING
+		if meta, class, err := m.backups.RestoreBackup(context.Background(), className, storageName, ID); err != nil {
+			if meta != nil {
+				m.RestoreStatus = string(meta.Status)
+			} else {
+				m.RestoreStatus = models.SnapshotRestoreMetaStatusFAILED
+			}
+			m.RestoreError = err
+			return
+		} else {
+			classM := models.Class{}
+			if err := json.Unmarshal([]byte(class), &classM); err != nil {
+				m.RestoreStatus = models.SnapshotRestoreMetaStatusFAILED
+				m.RestoreError = err
+				return
+			}
+			err := m.AddClass(ctx, principal, &classM)
+			if err != nil {
+				m.RestoreStatus = models.SnapshotRestoreMetaStatusFAILED
+				m.RestoreError = err
+				return
+			}
+
+			m.RestoreStatus = string(models.SnapshotRestoreMetaStatusSUCCESS)
 
-	if meta, err := m.backups.RestoreBackup(ctx, className, storageName, ID); err != nil {
-		return nil, err
+		}
+	}(ctx, className, ID)
+
+	p := ""
+	status, err := m.backups.CreateBackupStatus(ctx, className, storageName, ID)
+	if err == nil {
+		p = status.Path
 	} else {
-		status := string(meta.Status)
-		return &models.SnapshotRestoreMeta{
-			ID:          ID,
-			StorageName: storageName,
-			Status:      &status,
-			Path:        meta.Path,
-		}, nil
+		p = err.Error()
```

Comment:
in case of error you are passing `Error` message to path?

Reply:
I guess we should fail the entire call if getting the status fails.  I'll add that.

Reply:
Although my thinking here was that failing to get the status in an async environment shouldn't return a failure for the start call.